src/octree/batch_iterator.rs

```diff
@@ -111,28 +111,38 @@ where
             position: self.position.split_off(0),
             layers,
         };
+        //println!("internal callback sending last batch...");
         (self.func)(point_data)
     }
 
-    fn push_point_and_callback(&mut self, point: Point) -> Result<()> {
-        self.push_point(point);
-        if self.position.len() == self.position.capacity() {
-            return self.callback();
+    fn push_point_and_callback<F2>(
```

Comment:
Could we rename the function to `batch_points()` or `push_points_and_callback()` since it no longer just consumes single points?

Reply:
Actually, if we are changing the signature anyway, what do you think about generalizing this to return a proper `Result`? Then we wouldn't need our own error type, which is not really an error anyway :)

Reply:
as discussed offline, pointstream is a private struct, returning a proper result can be posponed when needed